### ISSUE TITLE
bug: PR request blocked by Error 400 for google_cloudfunctions2_function

### ISSUE THREAD

**Issue by msampathkumar** [terraform-google-modules/terraform-docs-samples#486]
### TL;DR

TL;DR - PR blocked due to a test failed. Cloud Function Error.

### Observed behavior

Blocked PR: https://github.com/terraform-google-modules/terraform-docs-samples/pull/485

Terraform File: https://github.com/terraform-google-modules/terraform-docs-samples/blob/1ea28834510310126196331d5951da96a24bb047/functions/basic_audit_logs/main.tf

Error Logs: 
> Step #2 - "test-samples-0": TestSamples/0/basic_audit_logs 2023-08-28T12:36:47Z command.go:185: Error: Error creating function: googleapi: Error 400: Validation failed for trigger projects/ci-tf-samples-0-96df/locations/us-central1/triggers/gcf-function-032436: Invalid resource state for "": Permission denied while using the Eventarc Service Agent. If you recently started to use Eventarc, it may take a few minutes before all necessary permissions are propagated to the Service Agent. Otherwise, verify that it has Eventarc Service Agent role.
Step #2 - "test-samples-0": TestSamples/0/basic_audit_logs 2023-08-28T12:36:47Z command.go:185: 
Step #2 - "test-samples-0": TestSamples/0/basic_audit_logs 2023-08-28T12:36:47Z command.go:185:   with google_cloudfunctions2_function.default,
Step #2 - "test-samples-0": TestSamples/0/basic_audit_logs 2023-08-28T12:36:47Z command.go:185:   on main.tf line 91, in resource "google_cloudfunctions2_function" "default":
Step #2 - "test-samples-0": TestSamples/0/basic_audit_logs 2023-08-28T12:36:47Z command.go:185:   91: resource "google_cloudfunctions2_function" "default" {
Step #2 - "test-samples-0": TestSamples/0/basic_audit_logs 2023-08-28T12:36:47Z command.go:185: 
Step #2 - "test-samples-0": TestSamples/0/basic_audit_logs 2023-08-28T12:36:47Z retry.go:144: 'terraform [apply -input=false -auto-approve -no-color -lock=false]' failed with the error 'error while running command: exit status 1; 
Step #2 - "test-samples-0": Error: Error creating function: googleapi: Error 400: Validation failed for trigger projects/ci-tf-samples-0-96df/locations/us-central1/triggers/gcf-function-032436: Invalid resource state for "": Permission denied while using the Eventarc Service Agent. If you recently started to use Eventarc, it may take a few minutes before all necessary permissions are propagated to the Service Agent. Otherwise, verify that it has Eventarc Service Agent role.
Step #2 - "test-samples-0": 
Step #2 - "test-samples-0":   with google_cloudfunctions2_function.default,
Step #2 - "test-samples-0":   on main.tf line 91, in resource "google_cloudfunctions2_function" "default":
Step #2 - "test-samples-0":   91: resource "google_cloudfunctions2_function" "default" {
Step #2 - "test-samples-0": ' but this error was expected and warrants a retry. Further details: Eventarc Service Agent IAM is eventually consistent
Step #2 - "test-samples-0": 
Step #2 - "test-samples-0": TestSamples/0/basic_audit_logs 2023-08-28T12:36:47Z retry.go:103: terraform [apply -input=false -auto-approve -no-color -lock=false] returned an error: error while running command: exit status 1; 
Step #2 - "test-samples-0": Error: Error creating function: googleapi: Error 400: Validation failed for trigger projects/ci-tf-samples-0-96df/locations/us-central1/triggers/gcf-function-032436: Invalid resource state for "": Permission denied while using the Eventarc Service Agent. If you recently started to use Eventarc, it may take a few minutes before all necessary permissions are propagated to the Service Agent. Otherwise, verify that it has Eventarc Service Agent role.
Step #2 - "test-samples-0": 
Step #2 - "test-samples-0":   with google_cloudfunctions2_function.default,
Step #2 - "test-samples-0":   on main.tf line 91, in resource "google_cloudfunctions2_function" "default":
Step #2 - "test-samples-0":   91: resource "google_cloudfunctions2_function" "default" {
Step #2 - "test-samples-0": . Sleeping for 1m0s and will try again.

Complete Error Logs [link](https://pantheon.corp.google.com/cloud-build/builds/d63e1c59-a19e-44c6-8cf6-1ae653555cc5;step=4?project=cloud-foundation-cicd)

### Terraform Configuration

```hcl
NA
```

### Terraform Version

```sh
NA
```

**Comment by msampathkumar** [comment]
As expected, this was one time issues.

PR https://github.com/terraform-google-modules/terraform-docs-samples tests are running successfully.
